source/components/molecules/CaseCard/test/CaseCard.test.tsx

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import { fireEvent } from "@testing-library/react-native";
+import { render } from "../../../../../test-utils";
+
+import CaseCard from "..";
+
+const mockTitle = "mockTitle";
+const mockButtonText = "mockButtonText";
+
+it("calls the callback function on card click", () => {
+  const mockCallback = jest.fn();
+
+  const { getByText } = render(
+    <CaseCard title={mockTitle} onCardClick={mockCallback} />
+  );
+
+  const titleElement = getByText(mockTitle);
+  fireEvent.press(titleElement);
+
+  expect(mockCallback).toHaveBeenCalled();
+});
+
+it("calls the callback function on button click", () => {
```

Comment:
perhaps you should mention which button you are testing?

Reply:
Good idea. I have updated the description.